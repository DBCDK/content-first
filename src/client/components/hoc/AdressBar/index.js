import withTagsFromUrl, {
  tagsFromURL,
  addTag,
  removeTag
} from './withTagsFromUrl.hoc';

export {withTagsFromUrl, tagsFromURL, addTag, removeTag};
