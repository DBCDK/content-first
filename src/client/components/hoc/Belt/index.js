import withChildBelt from './withChildBelt.hoc';
import withStoreBelt from './withStoreBelt.hoc';

export {withChildBelt, withStoreBelt};
