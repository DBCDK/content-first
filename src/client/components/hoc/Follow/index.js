import withFollow from './withFollow.hoc';
import withFollows from './withFollows.hoc';

export {withFollow, withFollows};
