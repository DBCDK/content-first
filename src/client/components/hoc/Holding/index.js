import withHoldings from './withHoldings.hoc';

export {withHoldings};
