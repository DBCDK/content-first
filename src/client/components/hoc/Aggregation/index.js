import {withListAggregation} from './withListAggregation.hoc';

export {withListAggregation};
