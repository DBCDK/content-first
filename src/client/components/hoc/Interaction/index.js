import {withInteractions} from './withInteractions.hoc';

export {withInteractions};
