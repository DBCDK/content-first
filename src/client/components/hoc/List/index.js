import {withLists} from './withLists.hoc';
import {withList} from './withList.hoc';
import {withListCreator} from './withListCreator.hoc';

export {withListCreator, withList, withLists};
