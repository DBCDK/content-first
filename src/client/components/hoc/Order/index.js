import {withOrders} from './withOrders.hoc';

export {withOrders};
