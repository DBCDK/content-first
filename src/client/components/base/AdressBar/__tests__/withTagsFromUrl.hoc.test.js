import React from 'react';
import renderer from 'react-test-renderer';

import {tagsFromURL, toggleTag} from '../withTagsFromUrl.hoc';
const filtercards = {
  Tempo: {
    title: 'Tempo',
    image: 'img/filters/speed.jpg',
    template: 'CardRange',
    icon: 'signal_cellular_4_bar',
    range: [5629, 5630, 5631, 5632, 5633],
    closeOnSelect: false,
    expanded: false,
    show: true
  }
};

describe('withTagsFromUrl.hoc', () => {
  it('expands single tag', () => {
    expect(tagsFromURL('test', filtercards)).toMatchSnapshot();
  });
  it('expands multiple tags', () => {
    expect(tagsFromURL(['test1', 'test2'], filtercards)).toMatchSnapshot();
  });
  it('expands a tag as a TAG', () => {
    expect(tagsFromURL(['5643'], filtercards)).toMatchSnapshot();
  });
  it('expands a tag range', () => {
    expect(tagsFromURL(['5629:5631'], filtercards)).toMatchSnapshot();
  });
  it('expands a pid as a TITLE', () => {
    expect(tagsFromURL(['870970-basis:123456'], filtercards)).toMatchSnapshot();
  });
  it('expands an author as a QUERY', () => {
    expect(tagsFromURL(['Peter Hansen'], filtercards)).toMatchSnapshot();
  });
  it('expands multiple tags of different kinds', () => {
    expect(
      tagsFromURL(['Peter Hansen', '870970-basis:123456', '5643'], filtercards)
    ).toMatchSnapshot();
  });

  it('removes the tag when it is already selected', () => {
    const expandedTags = tagsFromURL(
      ['Peter Hansen', '870970-basis:123456', '5643'],
      filtercards
    );
    expect(
      toggleTag(expandedTags, filtercards, '870970-basis:123456')
    ).toMatchSnapshot();
  });
  it('appends a tag when it is not already selected', () => {
    const expandedTags = tagsFromURL(
      ['Peter Hansen', '870970-basis:123456', '5643'],
      filtercards
    );
    expect(toggleTag(expandedTags, filtercards, 'Hest')).toMatchSnapshot();
  });
  it('appends a range tag when it is not already selected', () => {
    const expandedTags = tagsFromURL([], filtercards);
    expect(
      toggleTag(expandedTags, filtercards, [5629, 5631])
    ).toMatchSnapshot();
  });
  it('replace a range tag when it is already selected', () => {
    const expandedTags = tagsFromURL(
      ['5629:5630', '870970-basis:123456'],
      filtercards
    );
    expect(
      toggleTag(expandedTags, filtercards, [5629, 5631])
    ).toMatchSnapshot();
  });
  it('replace a range tag when it is already selected, and the selected tag is not in range format', () => {
    const expandedTags = tagsFromURL(
      ['5629:5630', '870970-basis:123456'],
      filtercards
    );
    expect(toggleTag(expandedTags, filtercards, 5631)).toMatchSnapshot();
  });
  it('removes a range tag when it is the full range', () => {
    const expandedTags = tagsFromURL(
      ['5629:5630', '870970-basis:123456'],
      filtercards
    );
    expect(
      toggleTag(expandedTags, filtercards, [5629, 5633])
    ).toMatchSnapshot();
  });
  it('removes a range tag when the exact same range is toggled', () => {
    const expandedTags = tagsFromURL(
      ['5629:5630', '870970-basis:123456'],
      filtercards
    );
    expect(
      toggleTag(expandedTags, filtercards, [5629, 5630])
    ).toMatchSnapshot();
  });
});
