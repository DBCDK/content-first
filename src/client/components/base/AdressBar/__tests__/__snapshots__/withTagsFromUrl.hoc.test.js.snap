// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withTagsFromUrl.hoc appends a range tag when it is not already selected 1`] = `
Array [
  Array [
    5629,
    5631,
  ],
]
`;

exports[`withTagsFromUrl.hoc appends a tag when it is not already selected 1`] = `
Array [
  "Peter Hansen",
  "870970-basis:123456",
  5643,
  "Hest",
]
`;

exports[`withTagsFromUrl.hoc expands a pid as a TITLE 1`] = `
Array [
  Object {
    "match": "870970-basis:123456",
    "pid": "870970-basis:123456",
    "type": "TITLE",
  },
]
`;

exports[`withTagsFromUrl.hoc expands a tag as a TAG 1`] = `
Array [
  Object {
    "id": 5643,
    "match": 5643,
    "parents": Array [
      "Stemning",
      "humoristisk",
    ],
    "score": 0,
    "title": "grotesk",
    "type": "TAG",
  },
]
`;

exports[`withTagsFromUrl.hoc expands a tag range 1`] = `
Array [
  Object {
    "fullRange": Array [
      5629,
      5630,
      5631,
      5632,
      5633,
    ],
    "inRange": Array [
      5629,
      5630,
      5631,
    ],
    "left": Object {
      "id": 5629,
      "parents": Array [
        "Tempo",
      ],
      "score": 0,
      "title": "statisk",
    },
    "match": Array [
      5629,
      5631,
    ],
    "right": Object {
      "id": 5631,
      "parents": Array [
        "Tempo",
      ],
      "score": 0,
      "title": "fremadskridende",
    },
    "type": "TAG_RANGE",
  },
]
`;

exports[`withTagsFromUrl.hoc expands an author as a QUERY 1`] = `
Array [
  Object {
    "match": "Peter Hansen",
    "query": "Peter Hansen",
    "type": "QUERY",
  },
]
`;

exports[`withTagsFromUrl.hoc expands multiple tags 1`] = `
Array [
  Object {
    "match": "test1",
    "query": "test1",
    "type": "QUERY",
  },
  Object {
    "match": "test2",
    "query": "test2",
    "type": "QUERY",
  },
]
`;

exports[`withTagsFromUrl.hoc expands multiple tags of different kinds 1`] = `
Array [
  Object {
    "match": "Peter Hansen",
    "query": "Peter Hansen",
    "type": "QUERY",
  },
  Object {
    "match": "870970-basis:123456",
    "pid": "870970-basis:123456",
    "type": "TITLE",
  },
  Object {
    "id": 5643,
    "match": 5643,
    "parents": Array [
      "Stemning",
      "humoristisk",
    ],
    "score": 0,
    "title": "grotesk",
    "type": "TAG",
  },
]
`;

exports[`withTagsFromUrl.hoc expands single tag 1`] = `
Array [
  Object {
    "match": "test",
    "query": "test",
    "type": "QUERY",
  },
]
`;

exports[`withTagsFromUrl.hoc removes a range tag when it is the full range 1`] = `
Array [
  "870970-basis:123456",
]
`;

exports[`withTagsFromUrl.hoc removes a range tag when the exact same range is toggled 1`] = `
Array [
  "870970-basis:123456",
]
`;

exports[`withTagsFromUrl.hoc removes the tag when it is already selected 1`] = `
Array [
  "Peter Hansen",
  5643,
]
`;

exports[`withTagsFromUrl.hoc replace a range tag when it is already selected 1`] = `
Array [
  Array [
    5629,
    5631,
  ],
  "870970-basis:123456",
]
`;

exports[`withTagsFromUrl.hoc replace a range tag when it is already selected, and the selected tag is not in range format 1`] = `
Array [
  Array [
    5629,
    5631,
  ],
  "870970-basis:123456",
]
`;
